roles/openstack-setup/tasks/users.yml

```diff
@@ -12,5 +12,5 @@
                     login_user=provider_admin
                     login_password={{ secrets.provider_admin_password }}
   with_items: keystone.services
-  when: endpoints[item.name] is defined and endpoints[item.name]
+  when: "{{item.name}}.enabled is undefined or {{item.name}}.enabled|bool"
```

Comment:
The logic change was made to use service enabled flag (like in other places). The check in place was always adding all endpoints even if services were not enabled since we define all endpoints in defaults file